aquadoggo/src/graphql/replication/context.rs

```diff
@@ -0,0 +1,191 @@
+        ids.into_iter()
+            .map(|id| {
+                let result = self.author_aliases.get(&id).map(|key| key.clone());
+                Ok(result)
+            })
+            .collect()
+    }
+
+    pub async fn entry_by_log_id_and_sequence<'a>(
+        &mut self,
+        log_id: LogId,
+        sequence_number: SequenceNumber,
+        author_alias: AuthorOrAlias,
+    ) -> Result<Option<SingleEntryAndPayload>> {
+        let author = self.get_author(author_alias)?;
+
+        let result = self
+            .entry_store
+            .get_entry_at_seq_num(&author.0, &log_id.0, &sequence_number.0)
+            .await?
+            .map(|entry| entry.into());
+
+        Ok(result)
+    }
+
+    pub async fn entry_by_hash<'a>(
+        &mut self,
+        hash: EntryHash,
+    ) -> Result<Option<SingleEntryAndPayload>> {
+        let result = self
+            .entry_store
+            .get_entry_by_hash(&hash.into())
+            .await?
+            .map(|entry_row| entry_row.into());
+
+        Ok(result)
+    }
+
+    pub async fn get_skiplinks<'a>(&mut self, entry: &Entry) -> Result<Vec<Entry>> {
+        let author = entry.as_ref().author();
+        let entry = decode_entry(entry.as_ref(), None)?;
+        let result = self
+            .entry_store
+            .get_certificate_pool(&author, entry.log_id(), entry.seq_num())
+            .await?
+            .into_iter()
+            .map(|entry| entry.entry_signed().clone().into())
+            .collect();
+
+        Ok(result)
+    }
+
+    pub async fn get_entries_newer_than_seq(
+        &mut self,
+        log_id: LogId,
+        author: AuthorOrAlias,
+        sequence_number: SequenceNumber,
+        max_number_of_entries: usize,
+    ) -> Result<Vec<EntryAndPayload>> {
+        let author = self.get_author(author)?;
+        let result = self
+            .entry_store
+            .get_paginated_log_entries(&author.0, &log_id.0, sequence_number.as_ref(), max_number_of_entries)
+            .await?
+            .into_iter()
+            .map(|entry| entry.into())
+            .collect();
+
+        Ok(result)
+    }
+
+    fn get_author(&mut self, author_alias: AuthorOrAlias) -> Result<PublicKey, anyhow::Error> {
+        let author = match author_alias {
+            AuthorOrAlias::Alias(alias) => self
+                .author_aliases
+                .get(&alias)
+                .ok_or(anyhow!(
+                    "author alias did not exist, you may need to re-alias your authors"
+                ))?
+                .clone(),
+            AuthorOrAlias::PublicKey(public_key) => public_key.clone(),
+        };
+        Ok(author)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::super::testing::MockEntryStore;
+    use super::Context;
+    use crate::graphql::replication::{
+        Author as GraphQLAuthor, LogId as GraphQLLogId, PublicKey, SequenceNumber,
+    };
+    use p2panda_rs::identity::Author;
+    use std::convert::TryInto;
```

Comment:
Mini nitpick: Import order should be `std`, external crates, `crate`, `super` :+1:

... aaaand there is a unnecessary newline on the top of the `tests` mod